In [1]:
import numpy as np
import seaborn as sns
import pandas as pd
import matplotlib.pyplot as plt
from scipy.stats import f_oneway

In [2]:
accident = pd.read_csv('Datasets//accident_data.csv')

In [3]:
accident


Index Accident_Severity Accident Date   Latitude  \
0       200701BS64157           Serious      5/6/2019  51.506187   
1       200701BS65737           Serious      2/7/2019  51.495029   
2       200701BS66127           Serious    26-08-2019  51.517715   
3       200701BS66128           Serious    16-08-2019  51.495478   
4       200701BS66837            Slight      3/9/2019  51.488576   
...               ...               ...           ...        ...   
660674  201091NM01760            Slight    18-02-2022  57.374005   
660675  201091NM01881            Slight    21-02-2022  57.232273   
660676  201091NM01935            Slight    23-02-2022  57.585044   
660677  201091NM01964           Serious    23-02-2022  57.214898   
660678  201091NM02142           Serious    28-02-2022  57.575210   

                   Light_Conditions           District Area  Longitude  \
0             Darkness - lights lit  Kensington and Chelsea  -0.209082   
1                          Daylight  Kensington and Chelsea  -0.173647   
2       Darkness - lighting unknown  Kensington and Chelsea  -0.210215   
3                          Daylight  Kensington and Chelsea  -0.202731   
4             Darkness - lights lit  Kensington and Chelsea  -0.192487   
...                             ...                     ...        ...   
660674                     Daylight                Highland  -3.467828   
660675       Darkness - no lighting                Highland  -3.809281   
660676                     Daylight                Highland  -3.862727   
660677       Darkness - no lighting                Highland  -3.823997   
660678                     Daylight                Highland  -3.895673   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
0                          1                   2                     Dry   
1                          1                   2             Wet or damp   
2                          1                   3                     Dry   
3                          1                   4                     Dry   
4                          1                   2                     Dry   
...                      ...                 ...                     ...   
660674                     2                   1                     Dry   
660675                     1                   1            Frost or ice   
660676                     1                   3            Frost or ice   
660677                     1                   2             Wet or damp   
660678                     1                   1             Wet or damp   

                 Road_Type Urban_or_Rural_Area     Weather_Conditions  \
0       Single carriageway               Urban     Fine no high winds   
1       Single carriageway               Urban  Raining no high winds   
2                      NaN               Urban                    NaN   
3       Single carriageway               Urban     Fine no high winds   
4                      NaN               Urban                    NaN   
...                    ...                 ...                    ...   
660674  Single carriageway               Rural     Fine no high winds   
660675  Single carriageway               Rural     Fine no high winds   
660676  Single carriageway               Rural     Fine no high winds   
660677  Single carriageway               Rural     Fine no high winds   
660678    Dual carriageway               Rural  Snowing no high winds   

                                Vehicle_Type  
0                                        Car  
1                                        Car  
2                      Taxi/Private hire car  
3       Bus or coach (17 or more pass seats)  
4                              Other vehicle  
...                                      ...  
660674                                   Car  
660675                                   Car  
660676                                   Car  
660677                 Motorcycle over 500cc  
660

In [5]:
accident.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 660679 entries, 0 to 660678
Data columns (total 14 columns):
 #   Column                   Non-Null Count   Dtype  
---  ------                   --------------   -----  
 0   Index                    660679 non-null  object 
 1   Accident_Severity        660679 non-null  object 
 2   Accident Date            660679 non-null  object 
 3   Latitude                 660654 non-null  float64
 4   Light_Conditions         660679 non-null  object 
 5   District Area            660679 non-null  object 
 6   Longitude                660653 non-null  float64
 7   Number_of_Casualties     660679 non-null  int64  
 8   Number_of_Vehicles       660679 non-null  int64  
 9   Road_Surface_Conditions  659953 non-null  object 
 10  Road_Type                656159 non-null  object 
 11  Urban_or_Rural_Area      660664 non-null  object 
 12  Weather_Conditions       646551 non-null  object 
 13  Vehicle_Type             660679 non-null  object 
dtypes: f

In [6]:
accident.describe()

Latitude      Longitude  Number_of_Casualties  Number_of_Vehicles
count  660654.000000  660653.000000         660679.000000       660679.000000
mean       52.553866      -1.431210              1.357040            1.831255
std         1.406922       1.383330              0.824847            0.715269
min        49.914430      -7.516225              1.000000            1.000000
25%        51.490690      -2.332291              1.000000            1.000000
50%        52.315641      -1.411667              1.000000            2.000000
75%        53.453452      -0.232869              1.000000            2.000000
max        60.757544       1.762010             68.000000           32.000000

In [7]:
accident.isnull().sum()

Index                          0
Accident_Severity              0
Accident Date                  0
Latitude                      25
Light_Conditions               0
District Area                  0
Longitude                     26
Number_of_Casualties           0
Number_of_Vehicles             0
Road_Surface_Conditions      726
Road_Type                   4520
Urban_or_Rural_Area           15
Weather_Conditions         14128
Vehicle_Type                   0
dtype: int64

In [8]:
accident.dtypes

Index                       object
Accident_Severity           object
Accident Date               object
Latitude                   float64
Light_Conditions            object
District Area               object
Longitude                  float64
Number_of_Casualties         int64
Number_of_Vehicles           int64
Road_Surface_Conditions     object
Road_Type                   object
Urban_or_Rural_Area         object
Weather_Conditions          object
Vehicle_Type                object
dtype: object

In [9]:
accident['Latitude'] = accident['Latitude'].fillna(accident['Latitude'].mode()[0])

In [10]:
accident.isnull().sum()

Index                          0
Accident_Severity              0
Accident Date                  0
Latitude                       0
Light_Conditions               0
District Area                  0
Longitude                     26
Number_of_Casualties           0
Number_of_Vehicles             0
Road_Surface_Conditions      726
Road_Type                   4520
Urban_or_Rural_Area           15
Weather_Conditions         14128
Vehicle_Type                   0
dtype: int64

In [11]:
accident['Longitude'] = accident['Longitude'].fillna(accident['Longitude'].mode()[0])

In [12]:
accident.isnull().sum()

Index                          0
Accident_Severity              0
Accident Date                  0
Latitude                       0
Light_Conditions               0
District Area                  0
Longitude                      0
Number_of_Casualties           0
Number_of_Vehicles             0
Road_Surface_Conditions      726
Road_Type                   4520
Urban_or_Rural_Area           15
Weather_Conditions         14128
Vehicle_Type                   0
dtype: int64

In [13]:
accident['Urban_or_Rural_Area'] = accident['Urban_or_Rural_Area'].fillna(accident['Urban_or_Rural_Area'].mode()[0])

In [14]:
accident.isnull().sum()

Index                          0
Accident_Severity              0
Accident Date                  0
Latitude                       0
Light_Conditions               0
District Area                  0
Longitude                      0
Number_of_Casualties           0
Number_of_Vehicles             0
Road_Surface_Conditions      726
Road_Type                   4520
Urban_or_Rural_Area            0
Weather_Conditions         14128
Vehicle_Type                   0
dtype: int64

In [15]:
accident['Road_Surface_Conditions'] = accident['Road_Surface_Conditions'].fillna(accident['Road_Surface_Conditions'].mode()[0])

In [16]:
accident.isnull().sum()

Index                          0
Accident_Severity              0
Accident Date                  0
Latitude                       0
Light_Conditions               0
District Area                  0
Longitude                      0
Number_of_Casualties           0
Number_of_Vehicles             0
Road_Surface_Conditions        0
Road_Type                   4520
Urban_or_Rural_Area            0
Weather_Conditions         14128
Vehicle_Type                   0
dtype: int64

In [17]:
accident['Road_Type'] = accident['Road_Type'].fillna(accident['Road_Type'].mode()[0])

In [18]:
accident.isnull().sum()

Index                          0
Accident_Severity              0
Accident Date                  0
Latitude                       0
Light_Conditions               0
District Area                  0
Longitude                      0
Number_of_Casualties           0
Number_of_Vehicles             0
Road_Surface_Conditions        0
Road_Type                      0
Urban_or_Rural_Area            0
Weather_Conditions         14128
Vehicle_Type                   0
dtype: int64

In [19]:
accident['Accident Date'] = pd.to_datetime(accident['Accident Date'], dayfirst = True, errors = 'coerce')

In [20]:
accident.dtypes

Index                              object
Accident_Severity                  object
Accident Date              datetime64[ns]
Latitude                          float64
Light_Conditions                   object
District Area                      object
Longitude                         float64
Number_of_Casualties                int64
Number_of_Vehicles                  int64
Road_Surface_Conditions            object
Road_Type                          object
Urban_or_Rural_Area                object
Weather_Conditions                 object
Vehicle_Type                       object
dtype: object

In [21]:
accident['Weather_Conditions'] = accident['Weather_Conditions'].fillna('unaccounted')

In [22]:
accident.isnull().sum()

Index                           0
Accident_Severity               0
Accident Date              395672
Latitude                        0
Light_Conditions                0
District Area                   0
Longitude                       0
Number_of_Casualties            0
Number_of_Vehicles              0
Road_Surface_Conditions         0
Road_Type                       0
Urban_or_Rural_Area             0
Weather_Conditions              0
Vehicle_Type                    0
dtype: int64

In [23]:
accident['Accident_Severity'] = accident['Accident_Severity'].astype('category')
accident['Light_Conditions'] = accident['Light_Conditions'].astype('category')
accident['District Area'] = accident['District Area'].astype('category')
accident['Road_Surface_Conditions'] = accident['Road_Surface_Conditions'].astype('category')
accident['Road_Type'] = accident['Road_Type'].astype('category')
accident['Urban_or_Rural_Area'] = accident['Urban_or_Rural_Area'].astype('category')
accident['Weather_Conditions'] = accident['Weather_Conditions'].astype('category')
accident['Vehicle_Type'] = accident['Vehicle_Type'].astype('category')



In [24]:
accident.dtypes

Index                              object
Accident_Severity                category
Accident Date              datetime64[ns]
Latitude                          float64
Light_Conditions                 category
District Area                    category
Longitude                         float64
Number_of_Casualties                int64
Number_of_Vehicles                  int64
Road_Surface_Conditions          category
Road_Type                        category
Urban_or_Rural_Area              category
Weather_Conditions               category
Vehicle_Type                     category
dtype: object

In [25]:
accident['Year'] = accident['Accident Date'].dt.year
accident['Month'] = accident['Accident Date'].dt.month
accident['Day'] = accident['Accident Date'].dt.day
accident['DayOfWeek'] = accident['Accident Date'].dt.dayofweek

accident.isnull().sum()

Index                           0
Accident_Severity               0
Accident Date              395672
Latitude                        0
Light_Conditions                0
District Area                   0
Longitude                       0
Number_of_Casualties            0
Number_of_Vehicles              0
Road_Surface_Conditions         0
Road_Type                       0
Urban_or_Rural_Area             0
Weather_Conditions              0
Vehicle_Type                    0
Year                       395672
Month                      395672
Day                        395672
DayOfWeek                  395672
dtype: int64

<h1>Univariate Analysis on accident that happened in 2019 </h1>

In [26]:
accident[accident['Year'] == 2019] 

Index Accident_Severity Accident Date   Latitude  \
0       200701BS64157           Serious    2019-06-05  51.506187   
1       200701BS65737           Serious    2019-07-02  51.495029   
4       200701BS66837            Slight    2019-09-03  51.488576   
6       200701BS67207           Serious    2019-09-05  51.501405   
7       200701BS67370             Fatal    2019-10-03  51.482260   
...               ...               ...           ...        ...   
182097       2.01E+12            Slight    2019-11-07  55.212275   
182101       2.01E+12            Slight    2019-12-02  55.009139   
182102       2.01E+12            Slight    2019-12-03  54.993061   
182103       2.01E+12           Serious    2019-12-06  55.127391   
182104       2.01E+12            Slight    2019-12-07  54.988202   

              Light_Conditions           District Area  Longitude  \
0        Darkness - lights lit  Kensington and Chelsea  -0.209082   
1                     Daylight  Kensington and Chelsea  -0.173647   
4        Darkness - lights lit  Kensington and Chelsea  -0.192487   
6                     Daylight  Kensington and Chelsea  -0.161577   
7        Darkness - lights lit  Kensington and Chelsea  -0.179486   
...                        ...                     ...        ...   
182097  Darkness - no lighting   Dumfries and Galloway  -3.413007   
182101                Daylight   Dumfries and Galloway  -3.319379   
182102                Daylight   Dumfries and Galloway  -3.131122   
182103   Darkness - lights lit   Dumfries and Galloway  -3.439964   
182104  Darkness - no lighting   Dumfries and Galloway  -3.122077   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
0                          1                   2                     Dry   
1                          1                   2             Wet or damp   
4                          1                   2                     Dry   
6                          1                   2                     Dry   
7                          3                   2                     Dry   
...                      ...                 ...                     ...   
182097                     1                   2                     Dry   
182101                     1                   1             Wet or damp   
182102                     1                   1            Frost or ice   
182103                     1                   2             Wet or damp   
182104                     1                   1             Wet or damp   

                 Road_Type Urban_or_Rural_Area     Weather_Conditions  \
0       Single carriageway               Urban     Fine no high winds   
1       Single carriageway               Urban  Raining no high winds   
4       Single carriageway               Urban            unaccounted   
6         Dual carriageway               Urban     Fine no high winds   
7       Single carriageway               Urban     Fine no high winds   
...                    ...                 ...                    ...   
182097    Dual carriageway               Rural     Fine no high winds   
182101  Single carriageway               Rural     Fine no high winds   
182102  Single carriageway               Rural     Fine no high winds   
182103  Single carriageway               Rural  Raining no high winds   
182104  Single carriageway               Rural     Fine no high winds   

                               Vehicle_Type    Year  Month  Day  DayOfWeek  
0                                       Car  2019.0    6.0  5.0        2.0  
1                                       Car  2019.0    7.0  2.0        1.0  
4                             Other vehicle  2019.0    9.0  3.0        1.0  
6       Van / Goods 3.5 tonnes mgw or under  2019.0    9.0  5.0        3.0  
7                                       Car  2019.0   10.0  3.0        3.0  
...                                     ...     ...    ...  ...        ...  
182097                                  C

In [27]:
df_2019 = accident[accident['Year'] == 2019] 

In [28]:
print(f"total number of accident in 2019 is {df_2019['Light_Conditions'].value_counts().sum()}") 
df_2019['Light_Conditions'].value_counts() 

total number of accident in 2019 is 71867


Light_Conditions
Daylight                       52518
Darkness - lights lit          14124
Darkness - no lighting          4242
Darkness - lighting unknown      719
Darkness - lights unlit          264
Name: count, dtype: int64

<h1>Insight No.1</h1>

    
<h2>Contrary to popular belief that most accidents happen in darkness or at night, most accidents happen in broad daylight. 
About 73.05% of the total accidents in 2019 happened during Daylight</h2>

In [29]:
df_2019['Weather_Conditions'].value_counts() 

Weather_Conditions
Fine no high winds       58345
Raining no high winds     7326
unaccounted               1635
Raining + high winds      1395
Other                     1373
Fine + high winds         1172
Snowing no high winds      349
Fog or mist                245
Snowing + high winds        27
Name: count, dtype: int64

<h1>Insight No. 2</h1>

<h2>Out of the total 71,867 accidents recorded, 81% (58,345) occurred under weather conditions described as "Fine with no high winds." This suggests that accidents remain highly prevalent even in seemingly favorable weather conditions. This highlights the need to explore other contributing factors such as driver behavior, road conditions, or traffic density that may play a significant role in accident occurrences</h2>

<h1>Q3. What are the top 5 vehicle types involved in accidents?</h1>

In [30]:
vehicle_type_counts = accident["Vehicle_Type"].value_counts()

In [31]:
vehicle_type_counts

Vehicle_Type
Car                                      497992
Van / Goods 3.5 tonnes mgw or under       34160
Bus or coach (17 or more pass seats)      25878
Motorcycle over 500cc                     25657
Goods 7.5 tonnes mgw and over             17307
Motorcycle 125cc and under                15269
Taxi/Private hire car                     13294
Motorcycle over 125cc and up to 500cc      7656
Motorcycle 50cc and under                  7603
Goods over 3.5t. and under 7.5t            6096
Other vehicle                              5637
Minibus (8 - 16 passenger seats)           1976
Agricultural vehicle                       1947
Pedal cycle                                 197
Data missing or out of range                  6
Ridden horse                                  4
Name: count, dtype: int64

<h1>Insight No.3</h1>
<h2>75.38% involve Cars, as they are the most common vehicle type.
5.17% involve Vans or small goods vehicles, which may have blind spots.
3.92% involve Buses or coaches, which carry many passengers.
3.88% involve Motorcycles over 500cc, which are high-risk due to lack of protection.
2.62% involve Large goods vehicles (7.5 tonnes or more), which take longer to stop.</h2>

<h1>Q4. How many accidents occur in each district area?</h1>

In [32]:
district_accidents = accident["District Area"].value_counts()

In [33]:
district_accidents

District Area
Birmingham            13491
Leeds                  8898
Manchester             6720
Bradford               6212
Sheffield              5710
                      ...  
Berwick-upon-Tweed      153
Teesdale                142
Shetland Islands        133
Orkney Islands          117
Clackmannanshire         91
Name: count, Length: 422, dtype: int64

<h1>Q5. Which road type has the highest number of accidents?</h1>

In [34]:
road_type_counts = accident["Road_Type"].value_counts()

In [35]:
road_type_counts

Road_Type
Single carriageway    496663
Dual carriageway       99424
Roundabout             43992
One way street         13559
Slip road               7041
Name: count, dtype: int64

<h1>Q6. What is the average number of vehicles involved in accidents?</h1>

In [36]:
average_vehicles = accident["Number_of_Vehicles"].mean()

In [37]:
average_vehicles

np.float64(1.8312554205597575)

<h1>Q7. What is the distribution of road surface conditions in accident cases?</h1>

In [38]:
road_surface_counts = accident["Road_Surface_Conditions"].value_counts()

In [39]:
road_surface_counts

Road_Surface_Conditions
Dry                     448547
Wet or damp             186708
Frost or ice             18517
Snow                      5890
Flood over 3cm. deep      1017
Name: count, dtype: int64

<h1>Q8. What is the distribution of accident severity levels?</h1>

In [40]:
severity_counts = accident["Accident_Severity"].value_counts

In [41]:
severity_counts

<bound method IndexOpsMixin.value_counts of 0         Serious
1         Serious
2         Serious
3         Serious
4          Slight
           ...   
660674     Slight
660675     Slight
660676     Slight
660677    Serious
660678    Serious
Name: Accident_Severity, Length: 660679, dtype: category
Categories (3, object): ['Fatal', 'Serious', 'Slight']>

<h1>Q9. Are accidents in rural areas more severe than in urban areas? </h1>

In [77]:
urban_rural_counts = accident["Urban_or_Rural_Area"].value_counts()

In [78]:
urban_rural_counts

Urban_or_Rural_Area
Urban          421678
Rural          238990
Unallocated        11
Name: count, dtype: int64

<h1>Q10. How is the number of casualties distributed across all accidents? </h1>

In [44]:
casualties_distribution = accident["Number_of_Casualties"].describe()

In [45]:
casualties_distribution

count    660679.000000
mean          1.357040
std           0.824847
min           1.000000
25%           1.000000
50%           1.000000
75%           1.000000
max          68.000000
Name: Number_of_Casualties, dtype: float64

<h1>Q11. Is there a relationship between road surface conditions and accident severity?</h1>

In [46]:
severity_road_surface = accident.groupby("Road_Surface_Conditions")["Accident_Severity"].value_counts()

C:\Users\GOD IS GOOD\AppData\Local\Temp\ipykernel_7004\2177308570.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  severity_road_surface = accident.groupby("Road_Surface_Conditions")["Accident_Severity"].value_counts()


In [47]:
severity_road_surface

Road_Surface_Conditions  Accident_Severity
Dry                      Slight               381049
                         Serious               61708
                         Fatal                  5790
Flood over 3cm. deep     Slight                  842
                         Serious                 152
                         Fatal                    23
Frost or ice             Slight                16317
                         Serious                2007
                         Fatal                   193
Snow                     Slight                 5290
                         Serious                 565
                         Fatal                    35
Wet or damp              Slight               160303
                         Serious               23785
                         Fatal                  2620
Name: count, dtype: int64

<h1>Q12. How do weather conditions affect the number of casualties?</h1>

In [48]:
weather_casualties = accident.groupby("Weather_Conditions")["Number_of_Casualties"].mean()

C:\Users\GOD IS GOOD\AppData\Local\Temp\ipykernel_7004\2013573567.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  weather_casualties = accident.groupby("Weather_Conditions")["Number_of_Casualties"].mean()


In [49]:
weather_casualties

Weather_Conditions
Fine + high winds        1.386018
Fine no high winds       1.350480
Fog or mist              1.452948
Other                    1.354869
Raining + high winds     1.416641
Raining no high winds    1.408214
Snowing + high winds     1.418079
Snowing no high winds    1.341776
unaccounted              1.233720
Name: Number_of_Casualties, dtype: float64

<h1>Q13. How does accident severity vary based on light conditions?</h1>

In [50]:
severity_light = accident.groupby("Light_Conditions")["Accident_Severity"].value_counts()

C:\Users\GOD IS GOOD\AppData\Local\Temp\ipykernel_7004\2800142113.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  severity_light = accident.groupby("Light_Conditions")["Accident_Severity"].value_counts()


In [51]:
severity_light

Light_Conditions             Accident_Severity
Darkness - lighting unknown  Slight                 5622
                             Serious                 794
                             Fatal                    68
Darkness - lights lit        Slight               108345
                             Serious               19130
                             Fatal                  1860
Darkness - lights unlit      Slight                 2138
                             Serious                 360
                             Fatal                    45
Darkness - no lighting       Slight                28651
                             Serious                7174
                             Fatal                  1612
Daylight                     Slight               419045
                             Serious               60759
                             Fatal                  5076
Name: count, dtype: int64

<h1>Q14. Does the number of vehicles involved impact accident severity?</h1>
<h2>Insight 14 </h2>

In [52]:
vehicles_severity = accident.groupby("Accident_Severity")["Number_of_Vehicles"].mean()

C:\Users\GOD IS GOOD\AppData\Local\Temp\ipykernel_7004\2263754706.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  vehicles_severity = accident.groupby("Accident_Severity")["Number_of_Vehicles"].mean()


In [53]:
vehicles_severity 

Accident_Severity
Fatal      1.786976
Serious    1.678327
Slight     1.855864
Name: Number_of_Vehicles, dtype: float64

<h1>Q15. What is the relationship between urban/rural areas and accident severity?</h1>
<h2> Insight 15 --While more accidents occur in urban areas due to higher traffic volume, rural accidents tend to be more severe, likely due to higher speeds, limited emergency response time, and more dangerous road conditions.</h2>

In [54]:
urban_rural_severity = accident.groupby("Urban_or_Rural_Area")["Accident_Severity"].value_counts()

C:\Users\GOD IS GOOD\AppData\Local\Temp\ipykernel_7004\1016623407.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  urban_rural_severity = accident.groupby("Urban_or_Rural_Area")["Accident_Severity"].value_counts()


In [55]:
urban_rural_severity 

Urban_or_Rural_Area  Accident_Severity
Rural                Slight               196077
                     Serious               37312
                     Fatal                  5601
Unallocated          Slight                   10
                     Serious                   1
                     Fatal                     0
Urban                Slight               367714
                     Serious               50904
                     Fatal                  3060
Name: count, dtype: int64

<h1>Q16. What combination of weather, road surface, and light conditions leads to severe accidents?</h1>
<h2>Insight 16 --Poor lighting and wet or flooded roads increase accident severity.</h2>

In [56]:
severe_conditions = accident[accident["Accident_Severity"] == "Fatal"].groupby(["Weather_Conditions", "Road_Surface_Conditions", "Light_Conditions"]).size().sort_values()

C:\Users\GOD IS GOOD\AppData\Local\Temp\ipykernel_7004\425524000.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  severe_conditions = accident[accident["Accident_Severity"] == "Fatal"].groupby(["Weather_Conditions", "Road_Surface_Conditions", "Light_Conditions"]).size().sort_values()


In [57]:
severe_conditions

Weather_Conditions     Road_Surface_Conditions  Light_Conditions           
Fine + high winds      Dry                      Darkness - lighting unknown       0
Raining no high winds  Dry                      Darkness - lighting unknown       0
                                                Darkness - lights unlit           0
                                                Daylight                          0
                       Flood over 3cm. deep     Darkness - lighting unknown       0
                                                                               ... 
                       Wet or damp              Daylight                        411
Fine no high winds     Wet or damp              Daylight                        584
                       Dry                      Darkness - no lighting          810
                                                Darkness - lights lit          1039
                                                Daylight                       3677


<h1>Q17. How do accident severity, number of casualties, and vehicle type relate to each other?</h1>
<h2>Insight 17 --Larger vehicles & two-wheelers have a higher risk of casualties per accident. </h2>

In [58]:
vehicle_casualties = accident.groupby("Vehicle_Type")["Number_of_Casualties"].mean().sort_values()

C:\Users\GOD IS GOOD\AppData\Local\Temp\ipykernel_7004\3339994007.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  vehicle_casualties = accident.groupby("Vehicle_Type")["Number_of_Casualties"].mean().sort_values()


In [59]:
vehicle_casualties

Vehicle_Type
Ridden horse                             1.250000
Motorcycle 125cc and under               1.332635
Motorcycle 50cc and under                1.337235
Other vehicle                            1.340075
Agricultural vehicle                     1.342065
Motorcycle over 125cc and up to 500cc    1.343521
Minibus (8 - 16 passenger seats)         1.345648
Bus or coach (17 or more pass seats)     1.349216
Goods 7.5 tonnes mgw and over            1.351881
Van / Goods 3.5 tonnes mgw or under      1.354537
Car                                      1.358841
Motorcycle over 500cc                    1.359434
Goods over 3.5t. and under 7.5t          1.362861
Taxi/Private hire car                    1.368663
Pedal cycle                              1.370558
Data missing or out of range             1.500000
Name: Number_of_Casualties, dtype: float64

<h1>Q18. Which district areas experience the highest number of casualties based on weather conditions?</h1>
<h2>Insight 18 --Most accidents with casualties occur in Birmingham with good weather conditions.</h2>

In [60]:
district_weather_casualties = accident.groupby(["District Area", "Weather_Conditions"])["Number_of_Casualties"].sum().sort_values()

C:\Users\GOD IS GOOD\AppData\Local\Temp\ipykernel_7004\1254624763.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  district_weather_casualties = accident.groupby(["District Area", "Weather_Conditions"])["Number_of_Casualties"].sum().sort_values()


In [61]:
district_weather_casualties

District Area      Weather_Conditions  
Caerphilly         Snowing + high winds        0
Chester-le-Street  Snowing + high winds        0
West Wiltshire     Snowing + high winds        0
Barrow-in-Furness  Fog or mist                 0
Middlesbrough      Other                       0
                                           ...  
Liverpool          Fine no high winds       6798
Manchester         Fine no high winds       6888
Bradford           Fine no high winds       7524
Leeds              Fine no high winds      10431
Birmingham         Fine no high winds      13957
Name: Number_of_Casualties, Length: 3798, dtype: int64

#

<h1>Q19. What are the most common conditions (weather, light, road type) for accidents involving multiple vehicles?</h1>
<h2>Insight 19 --Most accidents occur in good weather conditions during the day on single carriageways, indicating that traffic volume plays a larger role in accident frequency than poor weather.</h2>

In [64]:
multi_vehicle_conditions = accident[accident["Number_of_Vehicles"] > 1].groupby(["Weather_Conditions", "Light_Conditions", "Road_Type"]).size().sort_values()

C:\Users\GOD IS GOOD\AppData\Local\Temp\ipykernel_7004\2146238190.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  multi_vehicle_conditions = accident[accident["Number_of_Vehicles"] > 1].groupby(["Weather_Conditions", "Light_Conditions", "Road_Type"]).size().sort_values()


In [65]:
multi_vehicle_conditions

Weather_Conditions     Light_Conditions             Road_Type         
Snowing no high winds  Darkness - lights unlit      Slip road                  0
Snowing + high winds   Darkness - lighting unknown  One way street             0
Snowing no high winds  Darkness - no lighting       One way street             0
Fog or mist            Darkness - lighting unknown  Slip road                  0
Snowing no high winds  Darkness - no lighting       Roundabout                 0
                                                                           ...  
Fine no high winds     Daylight                     Roundabout             23751
Raining no high winds  Daylight                     Single carriageway     26703
Fine no high winds     Darkness - lights lit        Single carriageway     43099
                       Daylight                     Dual carriageway       44594
                                                    Single carriageway    214273
Length: 225, dtype: int64

<h2>Insight 20 --No correlation exists between Number of Casualties and the Number of Vehicles.</h2>

In [73]:
casualties_vehicles = accident['Number_of_Casualties'].corr(accident['Number_of_Vehicles'])
casualties_vehicles

np.float64(0.22888886126927635)